tokio/src/runtime/queue.rs

```diff
@@ -36,13 +36,13 @@ pub(super) struct Inject<T: 'static> {
 
 pub(super) struct Inner<T: 'static> {
     /// Concurrently updated by many threads.
-    head: AtomicU32,
+    head: AtomicU16,
 
     /// Only updated by producer thread but read by many threads.
-    tail: AtomicU32,
+    tail: AtomicU8,
```

Comment:
nit: it would be nice to explain how indices are packed into these fields here...